core/trino-main/src/main/java/io/trino/operator/project/PageFieldsToInputParametersRewriter.java

```diff
@@ -42,77 +45,113 @@ private PageFieldsToInputParametersRewriter() {}
     public static Result rewritePageFieldsToInputParameters(RowExpression expression)
     {
         Visitor visitor = new Visitor();
-        RowExpression rewrittenProjection = expression.accept(visitor, null);
-        InputChannels inputChannels = new InputChannels(visitor.getInputChannels());
+        RowExpression rewrittenProjection = expression.accept(visitor, true);
+        InputChannels inputChannels = new InputChannels(visitor.getInputChannels(), visitor.getEagerlyLoadedChannels());
         return new Result(rewrittenProjection, inputChannels);
     }
 
     private static class Visitor
-            implements RowExpressionVisitor<RowExpression, Void>
+            implements RowExpressionVisitor<RowExpression, Boolean>
     {
         private final Map<Integer, Integer> fieldToParameter = new HashMap<>();
         private final List<Integer> inputChannels = new ArrayList<>();
```

Comment:
why `inputChannels` is a list while `eagerlyLoadedChannels` is a set?

Reply:
it was indeed unnecessary earlier, in the updated code using set is needed